macros/_assertions_expression.sql

```diff
@@ -117,3 +117,30 @@ ARRAY_DISTINCT(ARRAY(
 )) AS {{ column }}
 
 {%- endmacro %}
+
+{%- macro redshift___assertions_expression(column, assertions) -%}
+
+ARRAY_FLATTEN(
+    ARRAY(
+        {%- for assertion_id, assertion_config in assertions.items() %}
+        {%- set expression =
+            assertion_config.expression
+            if '\n' not in assertion_config.expression
+            else assertion_config.expression | indent(12) -%}
+        {%- set description= assertion_config.description -%}
+        {%- set null_as_exception =
+            'FALSE'
+            if (assertion_config.null_as_exception is not defined
+                or assertion_config.null_as_exception is true)
+            else 'TRUE' %}
+        /* {{ assertion_id }}: {{ description }} */
+        CASE WHEN COALESCE({{ expression }}, {{ null_as_exception }}) = FALSE
+            THEN ARRAY('{{ assertion_id }}')
+            ELSE ARRAY()
+        END,
```

Comment:
Why not with a `IF` statement like the following ? It is jsut to stay coherent with other functions but it is ok to keep it at is if needed :)

```sql
        /* {{ assertion_id }}: {{ description }} */
        IF(
            COALESCE({{ expression }}, {{ null_as_exception }}) = FALSE,
            ARRAY('{{ assertion_id }}'),
            ARRAY()
        ),
```